Makefile

```diff
@@ -33,19 +33,25 @@ T_RESET := \e[0m
 .PHONY: all
 all: 1.20 1.21 1.22 1.23 ## Build all versions of EKS Optimized AL2 AMI
 
+# ensure that these flags are equivalent to the rules in the .editorconfig
+SHFMT_FLAGS := --list \
+--language-dialect auto \
+--indent 2 \
+--binary-next-line \
+--case-indent \
+--space-redirects \
+--keep-padding
+
 .PHONY: fmt
 fmt: ## Format the source files
-	# ensure that these flags are equivalent to the rules in the .editorconfig
-	shfmt \
-	  --list \
-	  --write \
-	  --language-dialect auto \
-	  --indent 2 \
-	  --binary-next-line \
-	  --case-indent \
-	  --space-redirects \
-	  --keep-padding \
-	  $(MAKEFILE_DIR)
+	shfmt $(SHFMT_FLAGS) --write $(MAKEFILE_DIR)
+
+.PHONY: lint
+lint: ## Check the source files for syntax and format issues
+	shfmt  $(SHFMT_FLAGS) --diff $(MAKEFILE_DIR)
+	for FILE in $$(find $(MAKEFILE_DIR) -type f -name '*.sh'); do
+	  shellcheck --format gcc --severity error "$$FILE"
```

Comment:
I started with `error` severity, because there's only one of those right now; while there are 18 warnings. I can raise the bar once we address more of them (or decide whether a lower severity bar makes sense).